modules/conversion-center/base/widget-link-in-bio-base.php

```diff
@@ -390,17 +401,22 @@ protected function add_icons_controls(): void {
 			]
 		);
 
-		$this->add_control(
-			'custom_panel_alert',
-			[
-				'type' => Controls_Manager::ALERT,
-				'alert_type' => 'info',
-				'content' => sprintf(
-					__( 'Add up to <b>%d</b> icons', 'elementor' ),
-					5
-				),
-			]
-		);
+		do_action( 'elementor/conversion-center/content/controls/icons/before', $this );
```

Comment:
Why do we need this hook? Can we use `$this->start_injection()` instead?

Reply:
@louiswol94 not sure how `$this->start_injection()` would work.  I tried looking at examples but not 100% clear.
This is used to inject a control that could be here or could be in a different control section based on the variation, so I put the control in the trait and used an action.

Reply:
@nicoladj77 Sorry, not sure I understand. If the `do_action()` is here, the control will always be rendered here. How can it be here, or in a different control section?

Reply:
@nicoladj77 Maybe this is an example?
https://github.com/elementor/elementor-pro/blob/main/modules/carousel/widgets/media-carousel.php#L131

Reply:
@louiswol94 ok clear now.